### PR TITLE
Allow cross compiling from m1 to intel on Macos

### PR DESCRIPTION
**Pull request by hellozee**

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -264,6 +264,10 @@ def _build(self, dest_dir):
         if self.conanfile.settings.os == "Linux":
             env_vars["LDFLAGS"] += " -Wl,-rpath,'$$ORIGIN/../lib' -Wl,--disable-new-dtags"
         
+        # cause for some reason the CFLAGS are update but not the LDFLAGS
+        if self.conanfile.settings.os == "Macos" and self.conanfile.settings.arch == "x86_64":
+            env_vars["LDFLAGS"] += " -arch x86_64"
```
[the remark this answers is not in the document]
Comment (hellozee) [reply conanfile.py:269]: You mean something like this,
```
os=Macos
os_build=Macos
arch=armv8
arch_build=x86_64
compiler=clang
compiler.version=14
compiler.libcxx=libc++
build_type=Release
[options]
[build_requires]
[env]
```

[the remark this answers is not in the document]
Comment (dean0x7d) [reply conanfile.py:269]: That's the old way of doing it (Conan < v1.24). See here for the new way (2 profile files): https://docs.conan.io/en/latest/systems_cross_building/cross_building.html#conan-v1-24-and-newer

[the remark this answers is not in the document]
Comment (hellozee) [reply conanfile.py:269]: The old way didn't work (probably cause we don't use the variables in the recipe) but I don't see a way to leverage the new way into our existing pipeline

[the remark this answers is not in the document]
Comment (dean0x7d) [reply conanfile.py:269]: OK, then forget all that. Could you just ensure that macOS ARM on x86 works the same that you already set up x86 on ARM? Add an entry to the changelog and bump the patch version.

**Comment by hellozee** [comment]
arm only for mac, *intense facepalm*

**Comment by hellozee** [comment]
welp, this needs to be redone now

**Pull request closed without merging**